### PR TITLE
Refactor brushless support

### PR DESCRIPTION
**Pull request by jedahan**
> Note: This is a draft PR until I get back to my axidraw (~Sep 20)

This is based off the esbuild branch, and contains a mix of things.

Apologies for mixing a few different things, hopefully its a legible PR otherwise.

### Changes

* Add UI to frontend to choose brushless hardware
* Add new number `minPenPosition` to Plan, which I think is different for each kind of hardware
  * This might be part of fixing the pen up/down not moving the full range on the brushless motor
* Re-adding the 'last move' pen up position logic in `Planning.plan`
  * I think this was removed during another rebase, not sure how intentional that was
* Broadcast the `com` path used in the server to all websockets (was hardcoded to `/dev/XXX` before) 

### Refactoring

* Move the `--hardware` cli option to be global, so it can be used in `saxi plot`.
* Switch `brushless` from a boolean to `hardware` enum
  * This just made it easier for me to refactor
* Add `hardware` to PlanOptions, so that `Planning.plan` works
  * There may be a way to remove `hardware` in the future with further refactoring
* A lot of small typescript and eslint cleanups, mostly harmless

### CODE REVIEW

File: src/cli.ts
Comment (jedahan) [inline src/cli.ts:1]: Main change is making the type of `--hardware` an enum. Everything else is just style.

File: build.mjs
Comment (jedahan) [inline build.mjs:1]: Esbuild migration

File: renovate.json
Comment (jedahan) [inline renovate.json:1]: esbuild

File: src/ebb.ts
Comment (jedahan) [inline src/ebb.ts:1]: Making `--hardware` an enum

**Review by jedahan — COMMENTED** [review]
Will continue giving file-by-file context soon

**Comment by jedahan** [comment]
@alexrudd2 mind rebasing your brushless branch on top of your main to include the esbuild stuff

**Comment by alexrudd2** [comment]
> @alexrudd2 mind rebasing your brushless branch on top of your main to include the esbuild stuff?

Ok, done.  Not tested, though :)

**Comment by alexrudd2** [comment]
I'm OK with all your changes generally, but this branch has gotten really unwieldy.

Please check out https://github.com/alexrudd2/saxi/pull/87, which clears up some of the merge conflicts.

Also, it would be easier to land a lot of the stuff (like the formatting, or splitting the ts-config) separately.

**Comment by jedahan** [comment]
Apologies for all the extra work I made, I am going to close this PR and work off the other one.

**Pull request closed without merging**